extensions/scarb-snforge-test-collector/src/compilation/test_collector.rs

```diff
@@ -149,23 +161,44 @@ pub fn collect_tests(
         })
         .collect_vec()
         .into_iter()
-        .map(|(test_name, config)| TestCaseRaw {
-            name: test_name,
-            available_gas: config.available_gas,
-            ignored: config.ignored,
-            expected_result: config.expected_result,
-            fork_config: config.fork_config,
-            fuzzer_config: config.fuzzer_config,
+        .map(|(test_name, config)| {
+            let test_details = build_test_details(&test_name, &sierra_program).unwrap();
+            TestCaseRaw {
+                name: test_name,
+                available_gas: config.available_gas,
+                ignored: config.ignored,
+                expected_result: config.expected_result,
+                fork_config: config.fork_config,
+                fuzzer_config: config.fuzzer_config,
+                test_details,
+            }
         })
         .collect();
 
-    let sierra_program = replace_sierra_ids_in_program(db, &sierra_program.program);
-
     validate_tests(sierra_program.clone(), &collected_tests)?;
 
     Ok((sierra_program, collected_tests))
 }
 
+fn build_test_details(test_name: &str, sierra_program: &Program) -> Result<TestDetails> {
+    let function_finder = FunctionFinder::new(sierra_program.clone())?;
+    let func = function_finder.find_function(test_name)?;
+
+    let type_size_map =
+        get_type_size_map(sierra_program, &function_finder.sierra_program_registry).unwrap();
```

Comment:
Why not create the function finder from the pair with some builder instead of making these two args public? Imho this obscures the relation between the two a bit.

Reply:
Is it safe to unwrap?